### PR TITLE
change cccid handling to target entire object and add print cert info in support of status action a la yubico-piv-tool. prints raw OIDs not friendly names at present.

### PR DESCRIPTION
**Pull request by carl-wallace**

### CODE REVIEW

File: src/cccid.rs [hunk 1]
```diff
@@ -48,16 +48,27 @@ const CCC_TMPL: &[u8] = &[
...
+    pub fn cccid(&self) -> Result<CccCardId, Error> {
+        let mut cccid = [0u8; YKPIV_CCCID_SIZE];
+        cccid.copy_from_slice(&self.0[CCC_ID_OFFS..(CCC_ID_OFFS + YKPIV_CCCID_SIZE)]);
+        Ok(CccCardId(cccid))
+    }
 
-impl CCCID {
     /// Generate a random CCCID
-    pub fn generate() -> Result<Self, Error> {
+    pub fn generate() -> Result<CccCardId, Error> {
```
Comment (tarcieri) [inline src/cccid.rs]: It seems weird for this method to not return `Self`

File: src/certificate.rs [hunk 2]
```diff
@@ -400,3 +401,49 @@ mod read_pki {
         }
     }
 }
+
+///Write information about certificate found in slot a la yubico-piv-tool output.
+pub fn print_cert_info(yubikey: &mut YubiKey, slot: SlotId) -> Result<(), Error> {
```
Comment (tarcieri) [inline src/certificate.rs]: This feels like it should be something like a `Debug` or `Display` impl, but offhand I'm not sure the right way to do that...

(or perhaps it should just exist in the CLI)

File: src/certificate.rs [hunk 3]
```diff
@@ -400,3 +401,49 @@ mod read_pki {
...
+        Err(e) => {
+            println!("error reading certificate in slot {:?}: {}", slot, e);
+            return Err(e);
+        }
+    };
+
+    if !buf.is_empty() {
+        let mut hasher = Sha256::new();
+        hasher.input(buf.clone().to_vec());
+        let fingerprint = hasher.result();
```
Comment (tarcieri) [inline src/certificate.rs]: You can accomplish the above with:

```suggestion
        let fingerprint = Sha256::digest(buf);
```

Comment (carl-wallace) [reply src/certificate.rs]: this worked: let fingerprint = Sha256::digest(&buf);

File: src/cccid.rs (hunk 1 above)
Comment (carl-wallace) [reply src/cccid.rs]: It's no longer the focus of the struct. It's like the fascn getter on the CHUID class.  I could see ditching CccCardId and returning an array

File: src/certificate.rs (hunk 2 above)
Comment (carl-wallace) [reply src/certificate.rs]: Just in the CLI is fine. I already wrote it in the CLI (after the certificate read method added in parallel was pointed out to me). Will resubmit minus this, with the Sha256::digest suggestion but not CccCardId suggestion.

**Pull request closed without merging**